shogun-lib/src/main/java/de/terrestris/shogun/lib/listener/LoginListener.java

```diff
@@ -40,7 +43,15 @@ public void onApplicationEvent(InteractiveAuthenticationSuccessEvent event) {
 
         // get user info from authentication object
         KeycloakPrincipal keycloakPrincipal = (KeycloakPrincipal) authentication.getPrincipal();
-        String keycloakUserId = keycloakPrincipal.getKeycloakSecurityContext().getIdToken().getSubject();
+        KeycloakSecurityContext keycloakSecurityContext = keycloakPrincipal.getKeycloakSecurityContext();
+        IDToken idToken = keycloakSecurityContext.getIdToken();
+        String keycloakUserId;
+        if (idToken != null) {
+            keycloakUserId = idToken.getSubject();
+        } else {
+            AccessToken accessToken = keycloakSecurityContext.getToken();
+            keycloakUserId = accessToken.getSubject();
+        }
```

Comment:
This looks *very* similiar to the changes/code in the `SecurityContextUtil`. Do you see any chance to reuse the code?

Reply:
Alright, see 57e97d8